src/scripts/core/models/column.ts

```diff
@@ -1,40 +1,100 @@
-            const randomNumber = Math.round(Math.random() * (9 - 1) + 1);
-
-            //const newToken = new Token(ran, this.columnID, availWidth, columnSize,);
-            //parentID: number, locationIndex: number, size: number, skIndex: number, availWidth: number
-            const newToken = new Token(this.columnID, i, this.columnSize, randomNumber, availWidth)
-            newToken.y = (availHeight / (columnSize/i));
+            const newToken = new Token(this.columnID, i, this.columnSize, availWidth, availHeight)
+            newToken.y = (availHeight / (this.columnSize/i));
 
             this.tokens.push(newToken);
             this.addChild(newToken);
         }
     }
 
-    public getAllTokens(): Token[] {
-        return this.tokens;
+    public processMatches(): void {
+            //move tokens from original positions in to the furthest matched token position
+            this.columnTweens();
+
+            //rebuild and reset the column's tokens
+            this.rebuildTokens();
+
+    }
+
+    private columnTweens(): void {
+        const nonMatchCombo: Token [] = [];
+        for(let columnInspector: number = 0; columnInspector < this.tokens.length-1; columnInspector++) {
+            const target = this.tokens[columnInspector];
+            if(!target.matched) {
+                nonMatchCombo.unshift(target);
+                for(let combo = columnInspector+1; combo <= this.tokens.length-1; combo++) {
+                    if(this.tokens[combo].matched) { 
+                        for(let i = 0; i < nonMatchCombo.length; i++){
+                            nonMatchCombo[i].moveTo(combo-i);
+                        }
+                    }
+                    else {
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    private rebuildTokens(): void {
+        const matchedTokens: Token [] = [];
+        const unmatchedTokens: Token [] = [];
+        for(let i = 0; i < this.tokens.length; i++) {
+            const target = this.tokens[i]
+            if(target.matched) {
+                target.hide();
+                target.interactive = false;
+                matchedTokens.push(target);
+            }
+
+            if(!target.matched) {
+                unmatchedTokens.push(target);
+            }
+        }
+        const newTokens = [...matchedTokens, ...unmatchedTokens]
+        for(var i = 0; i < this.columnSize; i++) {
+            newTokens[i]._verticalIndex = i;
+            newTokens[i].matched = false;
+        }
+        this.tokens = newTokens;
+    }
+
+    public hideMatches(): void {
+        this.tokens.forEach(token => {
+            if(token.matched) {
+                token.hide()
+            }
+        });
+    }
+
+    public revealMatches(): void {
+        this.tokens.forEach(token => {
+            if(token.matched) {
+                token.reveal()
+                token.matched = false;
+            }
+        });
+    }
```

Comment:
Just want to add a note about unexpected function behaviour, which I think can be seen here.

If I were a newbie to your code base and I called a function named revealMatches, I wouldn't expect it to also set token.matched to false. So I may end up introducing a bug or just generally being confused when something breaks.

Try not to add behaviour inside a function that isn't really related to that function. Or rather consider renaming the function so it doesn't appear to do one specific job while internally doing additional things. I can't see this function being called anywhere atm to confirm how it is intended to be used but it's something to keep in mind when grouping code together.